libnestutil/enum_bitfield.h

```diff
@@ -0,0 +1,127 @@
+/*
+ *  enum_bitfield.h
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef ENUM_BITFIELD_H
+#define ENUM_BITFIELD_H
+
+namespace nest
+{
+
+/**
+ * These methods support type-safe bitfields using C++'s enum classes.
+ *
+ * Define the bitfield flags as follows:
+ *
+ * .. code-block:: C++
+ *
+ *    enum class My_Flags : unsigned
+ *    {
+ *        FIRST_FLAG = 1 << 0,
```

Comment:
I could use any names for the flags here, correct?

Reply:
Correct, we could alternatively do `FOO_FLAG`, `XYZZY_FLAG`, ...